src/test/java/org/cactoos/experimental/TimedTest.java

```diff
@@ -0,0 +1,283 @@
+                Executors.newSingleThreadExecutor(),
+                1L,
+                TimeUnit.SECONDS,
+                () -> {
+                    throw new IllegalStateException("Something went wrong");
+                }
+            ).iterator().next(),
+            new Throws<>(
+                new StringContains("java.lang.IllegalStateException: Something went wrong"),
+                CompletionException.class
+            )
+        ).affirm();
+    }
+
+    /**
+     * Execute the tasks concurrently using {@link Timed} when
+     *  {@link ExecutorService} was initiated by {@link Timed} itself.
+     */
+    @Test
+    public void containsValuesWithInlineExecutorService() {
+        this.repeat(
+            arg -> new Assertion<>(
+                "Contains results from the callables when using the inline executor service",
+                new Timed<String>(
+                    TimedTest.THREADS,
+                    1L,
+                    TimeUnit.SECONDS,
+                    () -> {
+                        this.sleep();
+                        return TimedTest.FIRST_TEXT;
+                    },
+                    () -> {
+                        this.sleep();
+                        return TimedTest.SECOND_TEXT;
+                    },
+                    () -> {
+                        this.sleep();
+                        return TimedTest.THIRD_TEXT;
+                    }
+                ),
+                new HasValues<>(TimedTest.FIRST_TEXT, TimedTest.SECOND_TEXT, TimedTest.THIRD_TEXT)
+            ).affirm()
+        );
+    }
+
+    /**
+     * Execution takes longer than timeout when
+     *  {@link ExecutorService} was initiated by {@link Timed} itself.
+     */
+    @Test
+    public void failsDueToTimeoutWithInlineExecutorService() {
+        this.repeat(
+            arg -> new Assertion<>(
+                "Fails due to timeout when using the inline executor service",
+                () -> new Timed<String>(
+                    TimedTest.THREADS,
+                    1L,
+                    TimeUnit.MILLISECONDS,
+                    () -> {
+                        this.sleep();
+                        return TimedTest.FIRST_TEXT;
+                    },
+                    () -> {
+                        this.sleep();
+                        return TimedTest.SECOND_TEXT;
+                    },
+                    () -> {
+                        this.sleep();
+                        return TimedTest.THIRD_TEXT;
+                    }
+                ).iterator().next(),
+                new Throws<>(
+                    new IsEqual<>("java.util.concurrent.CancellationException"),
+                    CompletionException.class
+                )
+            ).affirm()
+        );
+    }
+
+    /**
+     * Repeat the test several times.
+     * @param test The test to execute.
+     */
+    private void repeat(final Proc<Object> test) {
+        final Object dummy = new Object();
```

Comment:
@paulodamaso also I've got an idea to create classes `RepeatedProc` and `RepeatedCallable` in order to get rid of this `dummy` objects. Also `Repeated` could be renamed to `RepeatedFunc`, but it'll break backward compatibility. 

I can create new issue or I can leave a puzzle in the scope of this issue. What do you think about it?

Reply:
@iakunin Yes, I think that we should have `Repeated` decorator for use with `Proc` and `Callable`. Leave a puzzle for this, please.

Reply:
@paulodamaso great! I've created a puzzle for this